### ISSUE TITLE
Announcement creation says date is "optional" but might actually be required

### ISSUE THREAD

**Issue by trwnh** [mastodon/mastodon#20791]
### Steps to reproduce the problem

1. Try to create an announcement
2. Add a start date, but no end date
3. Note that both "start" and "end" say "optional"
4. Try to submit

### Expected behaviour

Fields are optional

### Actual behaviour

Both start and end become required

### Specifications

main

**Comment by ramsey** [comment]
I've found (at least in Safari) that both fields are already populated, and there's no way to remove the date from the "optional" end field.

**Comment by iMagesh** [comment]
I checked it on chrome just now. Looks like date fields are not mandatory. I'm able to post an announcement with just the message leaving other fields empty. Maybe this is expected behavior?

Announcements without a date will be published immediately and might not be taken down automatically. Admins might have to unpublish it manually. This seems to be okay for me. I don't think it is a bug.

**Comment by trwnh** [comment]
> I'm able to post an announcement with just the message leaving other fields empty.

The issue is if you enter a start date, you must also enter an end date

**Comment by iMagesh** [comment]
> > I'm able to post an announcement with just the message leaving other fields empty.
> 
> The issue is if you enter a start date, you must also enter an end date

Still not sure if it is a bug. I can have a start date and not an end date if I want to end it manually, right?

**Comment by trwnh** [comment]
Not with the current form, you can't

**Comment by rixx** [comment]
Yup, same here – I wanted to only enter an end date, but apparently this makes the start date mandatory, which seems unintended.

**Comment by mjankowski** [comment]
The backend logic on this is indeed that if either of the start/end values are present, the other one becomes required.

Form looks like this as of latest release:

![Image](https://github.com/user-attachments/assets/d6e0ffc1-543a-4de4-ba1a-c4d2a493e2d1)

If you populate the start date field, the end field gets a red border to indicate required.

That said, I'm curious if a way to improve this is...

- JUST change the copy here? ie, remove "Optional" and/or otherwise tweak that language
- Tooltip with more verbose instructions?
- Link to docs with more verbose instructions?
- Some backend logic change?

Seems like the narrow original issue could be solved by removing "Optional", and maybe add'l tweaks.

**Comment by trwnh** [comment]
I would change the copy to remove "Optional", and then combine the two inputs into a single group -- this should make it more clear that the two input elements are linked, and if you fill one then you need to fill the other.

Something like

![Image](https://github.com/user-attachments/assets/0eecfee7-14b4-4314-8361-c20c5783472c)

One other thing that is confusing is that it's unclear how "time range" differs from "schedule". At the very least, the behavior of "announcement will be automatically unpublished at end_time" should probably be split out and grouped with "schedule publication".

![Image](https://github.com/user-attachments/assets/5d3c20c7-a524-4f04-b2c1-a2ba5f362354)

**Comment by mjankowski** [comment]
Yeah, there's sort of an implicit "is this announcement connected to some event or limited time window for whatever reason" question that the start/end are answering.

It might be useful to have data from servers that have been running long enough to have more than a few announcements, and see how many of them have any start/end data at all. If it's relatively rare, we could also change order here to move the start/end to a more obviously optional thing at end of form, after the required announcement text and (I'm guessing?) more frequently used scheduled value.

**Comment by Chewie9999** [comment]
This is still an issue on 4.5.3.
It's only optional if you don't specify _any_ date.
Also, if you tick "All day event", you still have to put a time in, even though the date should be sufficient.

It would be nice to be able to just specific an end date (and set "All day event"?) to set up a notification from "right now" to the end date, without having to add loads of details, which should not be necessary.
